erpc_c/port/erpc_threading_zephyr.cpp

```diff
@@ -53,14 +50,12 @@ void Thread::init(thread_entry_t entry, uint32_t priority, uint32_t stackSize, t
 void Thread::start(void *arg)
 {
     m_arg = arg;
-
-    erpc_assert(m_stack && "Set stack address");
-    k_thread_create(&m_thread, m_stack, m_stackSize, threadEntryPointStub, this, NULL, NULL, m_priority, 0, K_NO_WAIT);
+    m_thread_id = k_thread_create(&m_thread, saas_thread_stack, 4096, (k_thread_entry_t) threadEntryPointStub, this, NULL, NULL, m_priority, 0, K_NO_WAIT);
```

Comment:
One known issue here is that if more than one thread is allocated, then they will share the same stack space and will smash each other. Sadly, there is no good way to dynmically allocate a thread stack in zephyr. 

This isn't an issue for our current TCP server implementation, as that will only ever allocate one thread to handle TCP requests, but if we ever want to do anything more complex we will run into this problem.